File: fnc/examples/chapter10/naivemems.m
%%
% We'll examine the guessing/IVP approach for the case $\lambda=0.6$. 
lambda = 0.6;
phi = @(r,w,dwdr) lambda./w.^2 - dwdr./r;

%%
% We have to avoid $r=0$, or we would divide by zero.
a = eps;  b = 1;

%%
% We convert the ODE to a first-order system in $v_1$, $v_2$. 
f = @(r,v) [ v(2); phi(r,v(1),v(2)) ];

%%
% Now we try multiple guesses for the unknown $w(0)$ and plot the
% solutions. 
for w0 = 0.4:0.1:0.9
    [r,v] = ode45( f, [a,b], [w0;0] );
    plot(r,v(:,1)), hold on
end
xlabel('x'), ylabel('w(x)')    % ignore this line
title('Solutions for choices of w(0)')   % ignore this line

%%
% On the graph, it's the curve starting at $w(0)=0.8$ that comes closest
% to the required condition $w(1)=1$. 

File: julia/scripts/chapter10/naivemems.jl

lambda = 0.6
phi = (r,w,dwdr) -> lambda/w^2 - dwdr/r;

a = eps();  b = 1;

f = (v,p,r) -> [ v[2]; phi(r,v[1],v[2]) ];

using DifferentialEquations, Plots

for w0 = 0.4:0.1:0.9
    IVP = ODEProblem(f,[w0,0],(a,b))
    sol = solve(IVP)
    plot!(sol,vars=[1],label="w0 = $w0")
end
plot!([],[],label="",
    xaxis=("x"), yaxis=("w(x)"),   
    title="Solutions for choices of w(0)",leg=:bottomright)
